server/pbench/bin/index-pbench

```diff
@@ -520,15 +534,12 @@ def mk_result_data_actions(ptb, options, idx_prefix, idx_patterns, opctx):
     sources = rd.make_source()
     if not sources:
         return
+    template = idx_patterns["result-data"]["template"]
     for source, source_id in sources:
-        tstamp = rd.convert_timestamp(source['@timestamp'])
-        year, month, day = tstamp.split('T', 1)[0].split('-')[0:3]
+        idx_name = rd.generate_index_name(template, idx_prefix, source)
```

Comment:
We are generating an index name from the timestamp in `source` (in both versions - this is not a change made by this PR): is this correct? We can end up using the 2019-01 index, even though the experiment was run in 2018-12. Should we not be using the run_metadata timestamp to generate the index name? I have a feeling I asked you this before, but I don't remember.

Reply:
Well, for some of the code we were using the value from the `@timestamp` field (tool data and result data, I believe) and for other records we were using another value (have to look at run and toc docs to see what they were doing).

This change to use `generate_index_name()` is attempting to always make it self-consistent: we ONLY ever generate the timestamp we use in the index name from the source data's `@timestamp` field.